### ISSUE TITLE
Typescript Definitions

### ISSUE THREAD

**Issue by josh-sachs** [planttheidea/fast-equals#8]
I created the following typescript definition... i think it could be added to the distribution as index.d.ts to the npm package root to add Typescript Support. 

```
declare interface Comparator {
    (objectA, objectB): boolean;
}

declare interface Pairs {
    keys: any[];
    values: any[];
}

declare type createComparator = (createIsEqual) => Comparator;
declare type createIsSameValueZero = () => Comparator;
declare type toPairs = (iterable) => Pairs;
declare type areIterablesEqual = (objectA: any, objectB: any, comparator: Comparator, shouldCompareKeys: boolean) => boolean;

declare type fe = {
    createCustomEqual: createComparator;
    deep: Comparator;
    shallowEqual: Comparator;
}


export const createCustomEqual: createComparator;
export const deepEqual: Comparator;
export const shallowEqual: Comparator;
export default fe;

```

**Comment by planttheidea** [comment]
Awesome! Would you mind creating a PR for this? It would let you get credit as contributor, but also allow for a conventional merge setup. Thanks!

**Comment by josh-sachs** [comment]
PR Submitted.

**Comment by planttheidea** [comment]
Resolved with merging of #10. Thanks again!
